enginetest/testdata.go

```diff
@@ -39,428 +39,454 @@ func includeTable(includedTables []string, tableName string) bool {
 	return false
 }
 
+func wrapInTransaction(t *testing.T, db sql.Database, harness Harness, fn func()) {
+	if tdb, ok := db.(sql.TransactionDatabase); ok {
```

Comment:
we shouldn't run this if it's not a `TransactionalDatabase`, we should `t.skip()` or panic

Reply:
You kinda have to... it's a no-op for databases that don't support transactions, but required for those that do

I'll add a comment